tests/test_skills/test_abstract_round_abci/test_base.py

```diff
@@ -914,6 +941,26 @@ def test_check_majority_possible_passes_when_vote_set_nonempty_and_check_passes(
         """
         AbstractRound.check_majority_possible({"voter": "item"}, 2)
 
+    def test_check_majority_possible_passes_when_payload_attributes_majority_match(
+        self,
+    ) -> None:
+        """
+        Check that 'check_majority_possible' passes when set of votes is non-empty
+        and the majority of the attribute values match.
+
+        The check passes because:
+        - the threshold is 3 (participants are 4)
+        - 3 voters have the same attribute value in their payload
+        """
+        AbstractRound.check_majority_possible(
+            {
+                "voter_1": DummyPayload("voter_1", 0),
+                "voter_2": DummyPayload("voter_2", 0),
+                "voter_3": DummyPayload("voter_3", 0),
+            },
+            4,
+        )
```

Comment:
Shouldn't this pass? We have 3/4 agents who have voted for the same thing, but we get:
```
packages.valory.skills.abstract_round_abci.base.ABCIAppException: cannot reach quorum=3, number of remaining votes=1, number of most voted item's votes=1
```

Reply:
yes. The error is here

https://github.com/valory-xyz/open-autonomy/blob/8ca6c61ccc810e47f6dc1731bcda3257be63486e/packages/valory/skills/abstract_round_abci/base.py#L1029

and should be 
`largest_nb_votes = vote_count.most_common()[0][1]`

Reply:
ok so I jumped the gun. The issue arised because of `__eq__`  here, which includes the sender attribute, and hence they do not compare equal

https://github.com/valory-xyz/open-autonomy/blob/8ca6c61ccc810e47f6dc1731bcda3257be63486e/packages/valory/skills/abstract_round_abci/base.py#L263-L272

solution thus is to access the .data attribute. 
`Counter([tuple(sorted(v.data)) for v in  votes_by_participant.values()])`

Please do **not** use the `__hash__` method like such, as it is scheduled to be removed
`Counter(map(hash, votes_by_participant.values()))`

Reply:
ticket is here https://github.com/valory-xyz/open-autonomy/issues/1003